From ddab30a7345799c3a2acd8a0827125621342d5f7 Mon Sep 17 00:00:00 2001
From: Mike Turquette <mturquette@linaro.org>
Date: Mon, 4 May 2015 15:10:39 -0700
Subject: [PATCH 5/6] sched: sched feature for cpu frequency selection

This patch introduces the SCHED_ENERGY_FREQ sched feature, which is
implemented using jump labels when SCHED_DEBUG is defined. It is
statically set to false when SCHED_DEBUG is not defined and thus
disabled by default.

Signed-off-by: Michael Turquette <mturquette@linaro.org>

trivial fix for 4.1-rc4

Signed-off-by: Koan-Sin Tan <freedom.tan@linaro.org>
---
 kernel/sched/fair.c     | 5 +++++
 kernel/sched/features.h | 6 ++++++
 2 files changed, 11 insertions(+)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index be707cf..087e663 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -4253,6 +4253,11 @@ static inline void hrtick_update(struct rq *rq)
 }
 #endif
 
+static inline bool sched_energy_freq(void)
+{
+	return sched_feat(SCHED_ENERGY_FREQ);
+}
+
 static bool cpu_overutilized(int cpu);
 
 /*
diff --git a/kernel/sched/features.h b/kernel/sched/features.h
index 92bc36e..29332c9 100644
--- a/kernel/sched/features.h
+++ b/kernel/sched/features.h
@@ -97,3 +97,9 @@ SCHED_FEAT(NUMA_RESIST_LOWER, false)
  * decisions optimizing for energy efficiency.
  */
 SCHED_FEAT(ENERGY_AWARE, false)
+
+/*
+ * Scheduler-driven CPU frequency selection aimed to save energy based on
+ * load tracking
+ */
+SCHED_FEAT(SCHED_ENERGY_FREQ, false)
-- 
1.9.1

